hibernate-core/src/main/java/org/hibernate/query/sqm/tree/predicate/SqmComparisonPredicate.java

```diff
@@ -40,6 +40,13 @@ public SqmComparisonPredicate(
 		rightHandExpression.applyInferableType( expressableType );
 	}
 
+	private SqmComparisonPredicate(SqmComparisonPredicate affirmativeForm) {
+		super( true, affirmativeForm.nodeBuilder() );
+		this.leftHandExpression = affirmativeForm.leftHandExpression;
+		this.rightHandExpression = affirmativeForm.rightHandExpression;
+		this.operator = affirmativeForm.operator;
+	}
```

Comment:
Shouldn't this rather be?

```suggestion
	private SqmComparisonPredicate(SqmComparisonPredicate affirmativeForm) {
		super( false, affirmativeForm.nodeBuilder() );
		this.leftHandExpression = affirmativeForm.leftHandExpression;
		this.rightHandExpression = affirmativeForm.rightHandExpression;
		this.operator = affirmativeForm.operator.negated();
	}
```

Reply:
No.  The problem comes from the `#isNegated` call.  Your form reports that "incorrectly".  I quote "incorrectly" because I personally think the TCK very much oversteps the spec for some of these checks.

Reply:
In that case, shouldn't this be the solution then?

Shouldn't this rather be?

```suggestion
	private SqmComparisonPredicate(SqmComparisonPredicate affirmativeForm) {
		super( !affirmativeForm.isNegated(), affirmativeForm.nodeBuilder() );
		this.leftHandExpression = affirmativeForm.leftHandExpression;
		this.rightHandExpression = affirmativeForm.rightHandExpression;
		this.operator = affirmativeForm.operator;
	}
```

Reply:
The new node is always the negated.  `not(not(true)`.  Each `not` in-and-of-itself is negated form of the wrapped node